packages/components/src/components/PhoneInput/PhoneInput.tsx

```diff
@@ -0,0 +1,368 @@
+];
+
+export type PhoneInputProps = {
+  /**
+   * Custom id for gruppen ved behov. setter `${id}-country-code` som `id` for landskode og `${id}-phone-number` for telefonnummer.
+   */
+  id?: string;
+  /**
+   * Verdi bestående av landskode og telefonnummer.
+   * */
+  value?: PhoneInputValue;
+  /** Default verdi bestående av landkode og telefonnummer. */
+  defaultValue?: PhoneInputValue;
+  /**
+   * `onChange` funksjon.
+   */
+  onChange?: (value: PhoneInputValue) => void;
+  /** `name` attributt; setter `${name}-country-code` som `name` for landskode og `${name}-phone-number` for telefonnummer. */
+  name?: string;
+  /**
+   * Spesifiserer ved hvilket brekkpunkt og nedover versjonen for små skjermer skal vises; den stacker subkomponentene vertikalt.
+   */
+  smallScreenBreakpoint?: ScreenSizeLiteral;
+  /**
+   * Usynlig ledetekst for nedtrekksliste med landkoder. Brukes hvis default ikke passer eller ikke er beskrivende nok.
+   * @default "Landskode"
+   */
+  selectLabel?: string;
+  /**
+   * Ref for nedtrekksliste med landkoder.
+   */
+  selectRef?: ForwardedRef<HTMLSelectElement>;
+  /**
+   * Usynlig ledetekst for gruppen.  Brukes hvis default ikke passer eller ikke er beskrivende nok.
+   * @default "Landskode og telefonnummer"
+   */
+  groupLabel?: string;
+} & Pick<
+  InputProps,
+  | 'readOnly'
+  | 'disabled'
+  | 'width'
+  | 'componentSize'
+  | 'errorMessage'
+  | 'label'
+  | 'tip'
+  | 'required'
+  | 'className'
+  | 'style'
+  | 'aria-required'
+  | 'aria-describedby'
+>;
+
+export const PhoneInput = forwardRef<HTMLInputElement, PhoneInputProps>(
+  (
+    {
+      label,
+      readOnly,
+      errorMessage,
+      tip,
+      required,
+      width,
+      componentSize = 'medium',
+      name,
+      className,
+      style,
+      value,
+      selectLabel = 'Landskode',
+      selectRef,
+      onChange,
+      defaultValue,
+      'aria-required': ariaRequired,
+      'aria-describedby': ariaDescribedby,
+      groupLabel = 'Landskode og telefonnummer',
+      ...props
+    },
+    ref,
+  ) => {
+    const generatedId = useId();
+    const uniqueId = props.id ?? generatedId;
+    const phoneInputId = `${uniqueId}-phone-input`;
+    const phoneNumberId = `${uniqueId}-phone-number`;
+    const selectId = `${uniqueId}-country-code`;
+
+    const hasErrorMessage = !!errorMessage;
+    const hasTip = !!tip;
+    const hasLabel = !!label;
+    const hasMessage = hasErrorMessage || hasTip;
+
+    const tipId = derivativeIdGenerator(phoneInputId, 'tip');
+    const errorMessageId = derivativeIdGenerator(phoneInputId, 'errorMessage');
+
+    const [callingCode, setCallingCode] = useState('');
+    const [internalValue, setInternalValue] = useState<PhoneInputValue>({
+      countryCode: '',
+      phoneNumber: '',
+    });
+
+    useEffect(() => {
+      if (defaultValue) {
+        setInternalValue(defaultValue);
+      }
+    }, [defaultValue]);
+
+    const isControlled = value !== undefined;
```

Comment:
Out of scope i denne PRen, men kom til å tenke på at dette er nok noe vi gjør ganske mye. Controllable-komponenter med egen intern state. Kanskje en idé å ta ut denne logikken i de komponentene det gjelder i en `useControllableState`-hook, noe ala det Chakra gjør: https://github.com/chakra-ui/chakra-ui/blob/d0f502f1e60cf059072a0f806b268f8de9a630e8/packages/react/src/hooks/use-controllable-state.ts#L29

Reply:
Du har faktisk implementert den i Elsa, men jeg fikk den ikke til å funke 😅  kunne sett på det i egen PR.